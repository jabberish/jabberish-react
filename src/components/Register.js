import React from 'react';
import RegisterForm from '../containers/RegisterForm';
import { fetchRegister } from '../services/auth-api';

const Register = () => {

  const submitRegister = (user) => {
    fetchRegister(user)
      .then(res => {
        console.log(res);
      });
  };
  
  return (
    <>
      <h1>Register</h1>
      <RegisterForm submitRegister={submitRegister} />
    </>
  );
};

export default Register;
