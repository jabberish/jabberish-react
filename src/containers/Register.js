import React from 'react';

import AuthForm from '../components/AuthForm';

import { fetchRegister } from '../services/auth-api';

class Register extends React.Component {
  state = {
    username: '',
    password: '',
    registerSuccess: true,
    failMessage: 'That username is taken. Please try another!'
  }

  handleSubmit = e => {
    e.preventDefault();
    const user = {
      username: this.state.username,
      password: this.state.password
    };

    fetchRegister(user)
      .then(res => {
        if(res._id) {
          window.location = '/';
        } else {
          this.setState({ registerSuccess: false });
        }
      });
  };

  handleUpdate = e => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    return (
      <>
        <h1>Register</h1>
        <AuthForm 
          formTitle="Register"
          handleSubmit={this.handleSubmit}
          handleUpdate={this.handleUpdate}
          success={this.state.registerSuccess}
          failMessage={this.state.failMessage}
        />
      </>
    );
  }
}

export default Register;
