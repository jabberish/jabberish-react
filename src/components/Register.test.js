import React from 'react';
import { shallow } from 'enzyme';
import Register from './Register';

describe('Register component', () => {
  it('renders Register', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
