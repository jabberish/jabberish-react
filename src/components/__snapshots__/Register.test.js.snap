// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders Register 1`] = `
<Fragment>
  <h1>
    Register
  </h1>
  <RegisterForm />
</Fragment>
`;
